modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/dump/reader/ZipDumpReader.java

```diff
@@ -19,11 +22,15 @@
 import com.enonic.xp.dump.SystemLoadListener;
 import com.enonic.xp.repo.impl.dump.DefaultFilePaths;
 import com.enonic.xp.repo.impl.dump.PathRef;
+import com.enonic.xp.repo.impl.dump.RepoLoadException;
 import com.enonic.xp.repo.impl.dump.blobstore.ZipDumpReadBlobStore;
 
 public class ZipDumpReader
     extends AbstractDumpReader
 {
+
+    private static final Pattern ROOT_DUMP_DIR_PATTERN = Pattern.compile( "^([^/]+)\\/dump.json" );
```

Comment:
Missing `$` at the and. Without it `a/dump.jsons` would match, for ex. (this is why it is also better to use `matches`

Also `.` must be escaped

Reply:
The symbol `$` was added. I did not catch about `.`. What you mean?

Reply:
`dump.json` matches `dump?json` because `.` is any symbol.

Reply:
Got it.